vendor-extra/JatsContentBundle/tests/EventListener/BuildView/ExtLinkListenerTest.php

```diff
@@ -0,0 +1,189 @@
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertInstanceOf(TemplateView::class, $view);
+        $this->assertSame('@LiberoPatterns/link.html.twig', $view->getTemplate());
+        $this->assertEmpty($view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    public function nodeProvider() : iterable
+    {
+        yield 'no href' => ['<ext-link xmlns="http://jats.nlm.nih.gov">foo</ext-link>'];
+        yield 'relative link' => [
+            <<<XML
+<ext-link xlink:href="bar" xmlns="http://jats.nlm.nih.gov" xmlns:xlink="http://www.w3.org/1999/xlink">
+    foo
+</ext-link>
+XML
+            ,
+        ];
+        yield 'different namespace' => ['<ext-link xmlns="http://example.com">foo</ext-link>'];
+        yield 'different element' => ['<italic xmlns="http://jats.nlm.nih.gov">foo</italic>'];
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_is_not_the_link_template() : void
+    {
+        $listener = new ExtLinkListener($this->createFailingConverter());
+
+        $element = $this->loadElement(
+            <<<XML
+<ext-link xlink:href="bar" xmlns="http://jats.nlm.nih.gov" xmlns:xlink="http://www.w3.org/1999/xlink">
+    foo
+</ext-link>
+XML
+        );
+
+        $event = new BuildViewEvent($element, new TemplateView('template'));
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertInstanceOf(TemplateView::class, $view);
+        $this->assertSame('template', $view->getTemplate());
+        $this->assertEmpty($view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    /**
+     * @test
+     */
+    public function it_does_nothing_if_there_is_already_text_set() : void
+    {
+        $listener = new ExtLinkListener($this->createFailingConverter());
+
+        $element = $this->loadElement(
+            <<<XML
+<ext-link xlink:href="bar" xmlns="http://jats.nlm.nih.gov" xmlns:xlink="http://www.w3.org/1999/xlink">
+    foo
+</ext-link>
+XML
+        );
+
+        $event = new BuildViewEvent($element, new TemplateView('@LiberoPatterns/link.html.twig', ['text' => 'bar']));
+        $listener->onBuildView($event);
+        $view = $event->getView();
+
+        $this->assertInstanceOf(TemplateView::class, $view);
+        $this->assertSame('@LiberoPatterns/link.html.twig', $view->getTemplate());
+        $this->assertSame(['text' => 'bar'], $view->getArguments());
+        $this->assertEmpty($view->getContext());
+    }
+
+    /**
+     * @test
+     */
+    public function it_sets_the_text_argument_and_href_attribute() : void
+    {
+        $listener = new ExtLinkListener($this->createDumpingConverter());
+
+        $element = $this->loadElement(
+            <<<XML
+<jats:ext-link xlink:href="http://example.com" xmlns:jats="http://jats.nlm.nih.gov"
+    xmlns:xlink="http://www.w3.org/1999/xlink">
+    foo <jats:italic>bar</jats:italic> baz
```

Comment:
may be a complex example, maybe split:
- a test for text argument and href attribute
- a test with inline elements inside the content like `<jats:italic>bar</jats:italic>`